components/utils/useMixpanel.js

```diff
@@ -0,0 +1,39 @@
+import React, { useEffect, useState } from 'react'
+import { useRouter } from 'next/router'
+
+export default function useMixpanel() {
+  const MOOVWEB = 'moovweb.'
+  const RSF = 'reactstorefront.'
+
+  const { asPath } = useRouter()
+  const [mpId, setMpId] = useState(false)
+
+  useEffect(() => {
+    const mixpanelLoadInterval = setInterval(() => {
+      if (window.mixpanel_is_loaded) {
+        setMpId(window.mixpanel.get_distinct_id())
+        clearInterval(mixpanelLoadInterval)
+      }
+    }, 100)
+
+    return () => clearInterval(mixpanelLoadInterval)
+  }, [])
+
+  useEffect(() => {
+    if (mpId) {
+      const anchors = document.querySelectorAll('a')
```

Comment:
this useEffect could be called earlier than the render function of a child component(s). Meaning that mpId and asPath might never be added for child components?

Also does this collide with the other PR for mpid-s?.

Reply:
https://github.com/moovweb/le-deployer/pull/542